src/CommunityToolkit.Maui/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,177 @@
+	/// <remarks>Developers are still responsible for assigning the injected instance of <typeparamref name="TViewModel" /> 
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddScoped<TView, TViewModel>(this IServiceCollection services)
+		where TView : BindableObject
+		where TViewModel : class, INotifyPropertyChanged
+	{
+		return services
+			.AddScoped<TViewModel>()
+			.AddScoped<TView>();
+	}
+
+	/// <summary>
+	/// Adds a Page of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Scoped"/> lifetime
+	/// and registers a MAUI Shell route in <see cref="Routing"/> using the value of 
+	/// <paramref name="route"/> as the route.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="NavigableElement"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <param name="route">Route at which this page will be registered within Shell routing. 
+	/// Defaults to type name of <typeparamref name="TView" />.</param>
+	/// <param name="factory">RouteFactory to be used while creating the <see cref="NavigableElement"/> 
+	/// for the route. Defaults to TypeRouteFactory.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of <typeparamref name="TViewModel" /> 
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddScopedWithShellRoute<TView, TViewModel>(this IServiceCollection services, string? route = null, RouteFactory? factory = null)
+		where TView : NavigableElement
+		where TViewModel : class, INotifyPropertyChanged
+	{
+		RegisterShellRoute<TView>(route, factory);
+		return services.AddScoped<TView, TViewModel>();
+	}
+
+	/// <summary>
+	/// Registers routes in in <see cref="Routing"/> for Views registered using WithShellRouting methods.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="NavigableElement"/></typeparam>
+	/// <param name="route">Route at which this page will be registered within Shell routing. 
+	/// Defaults to type name of <typeparamref name="TView" />.</param>
+	/// <param name="factory">RouteFactory to be used while creating the <see cref="NavigableElement"/> 
+	/// for the route. Defaults to TypeRouteFactory.</param>
+	static void RegisterShellRoute<TView>(string? route = null, RouteFactory? factory = null)
+		where TView: NavigableElement
+	{
+		route ??= typeof(TView).Name;
```

Comment:
I believe that @brminnick doesn't like this here due to the .NET MAUI team recommendation to avoid any reflection during the application startup, this can impact. TBH I'm not sure if `typeof` is very bad here, do you have any input on that?

Reply:
Much to my disappointment, @brminnick and I discussed offline and I'm going to update the PR to make `string route` non-nullable which will remove the need for this line.